mappings/bnfdmd/data_downloader.py

```diff
@@ -0,0 +1,129 @@
+import glob
+import os
+import re
+from datetime import datetime
+from pathlib import Path
+from urllib.parse import unquote, urljoin, urlparse
+
+import requests
+import structlog
+from bs4 import BeautifulSoup
+
+
+logger = structlog.get_logger()
+
+
+# modelled after coding_systems/base/trud_utils.py::TrudDownloader
```

Comment:
nit: I'd consider referencing the commit ID, to make it easier for someone to piece together if the other code changes.